File: Manali/meta/EncodeandDecodeStrings.py
class Codec:
    def encode(self, strs: List[str]) -> str:
        """Encodes a list of strings to a single string.
        """
        res = ""
        for s in strs:
            res += str(len(s)) + "#" + s
        return res

    def decode(self, s: str) -> List[str]:
        """Decodes a single string to a list of strings.
        """
        ans = []
        i=0
        while i < len(s):
            temp = ""
            while s[i]!='#':
                temp+=s[i]
                i+=1
            temp = int(temp)
            ans.append(s[i+1:i+temp+1])
            i+=temp+1
        print(ans)
        return ans


TC: O(n)
SC: O(n)

class Codec:
    def encode(self, strs: List[str]) -> str:
        """Encodes a list of strings to a single string.
        """
        return "π".join(strs)

    def decode(self, s: str) -> List[str]:
        """Decodes a single string to a list of strings.
        """
        return s.split("π")

TC: O(n)
SC: O(k)

        


# Your Codec object will be instantiated and called as such:
# codec = Codec()
# codec.decode(codec.encode(strs))

File: parth/Week2/EncodeAndDecodeStrings.cpp
class Codec {
private:
    const string delimiter = "|";
public:

    // Encodes a list of strings to a single string.
    string encode(vector<string>& strs) {
        string result;
        for(string& str : strs){
            result.append(to_string(str.size()));
            result.append(delimiter);
            result.append(str);
        }
        return result;
    }

    // Decodes a single string to a list of strings.
    vector<string> decode(string s) {
        vector<string> result;
        for(int i=0; i<s.size();){
            int indexOfDelimiter = s.find(delimiter, i);
            int lenOfString = stoi(s.substr(i,indexOfDelimiter));
            result.push_back(s.substr(indexOfDelimiter+1,lenOfString));
            i=indexOfDelimiter+lenOfString+1;
        }
        return result;
    }
};

// Your Codec object will be instantiated and called as such:
// Codec codec;
// codec.decode(codec.encode(strs));